File: app/src/main/java/net/squanchy/service/repository/EventRepository.java
package net.squanchy.service.repository;

import java.util.List;

import net.squanchy.schedule.domain.view.Event;

import io.reactivex.Completable;
import io.reactivex.Observable;

public interface EventRepository {

    Observable<Event> event(String eventId, String userId);

    Observable<List<Event>> events(String userId);

    Completable addFavorite(String eventId, String userId);

    Completable removeFavorite(String eventId, String userId);
}

File: app/src/main/java/net/squanchy/service/repository/EventRepository.kt
package net.squanchy.service.repository

import io.reactivex.Completable
import io.reactivex.Observable
import net.squanchy.schedule.domain.view.Event

interface EventRepository {

    fun event(eventId: String, userId: String): Observable<Event>

    fun events(userId: String): Observable<List<Event>>

    fun addFavorite(eventId: String, userId: String): Completable

    fun removeFavorite(eventId: String, userId: String): Completable
}
